### ISSUE TITLE
Still can't send envoys

### ISSUE THREAD

**Issue by Oettinger77** [Azurency/CQUI_Community-Edition#355]
I still can't send envoys to city states. I thought this bug was already fixed.

Regards

**Comment by Azurency** [comment]
Hello, can you please respect the bug report template when posting an issue. 
Especially the part :
```Save game and logs : Please attach if you can your save game and the logs. You can find them in ..\documents\my games\sid meier's civilization vi\ on windows or /Users/[user]/Library/Application Support/Sid Meier's Civilization VI/ on Mac OS.```
Otherwise we can't really help you.

**Comment by angryansari** [comment]
[net_message_debug.log](https://github.com/Azurency/CQUI_Community-Edition/files/2400498/net_message_debug.log)


[CleanSave.Civ6Save.txt](https://github.com/Azurency/CQUI_Community-Edition/files/2400502/CleanSave.Civ6Save.txt)
not OP, but im having same issue. ive attached mine. which log specifically did you need?

**Comment by ihendriks** [comment]
> not OP, but im having same issue. ive attached mine. which log specifically did you need?
Just zip all of them.

The save you provided works perfectly fine for me. Are you using any other mods? If you do, try disabling those. Also please let me know the version of your game and CQUI mod, and what OS you are on.

**Comment by lautit** [comment]
Same thing here :(

[Lua.log](https://github.com/Azurency/CQUI_Community-Edition/files/2418352/Lua.log)

[AutoSave_0274.Civ6Save.txt](https://github.com/Azurency/CQUI_Community-Edition/files/2418357/AutoSave_0274.Civ6Save.txt)

![image](https://user-images.githubusercontent.com/14226299/46061218-5569dd80-c13c-11e8-9d5c-9b657c8c5eb1.png)

Any ideas?

**Comment by lautit** [comment]
I think I got it working. There is a conflict between the function `AddCityStateRow` in CQUI_Community-Edition\Assets\UI\PartialScreens and the one in \DLC\Expansion1\UI\Replacements.

A simple refactor:

**CQUI_Community-Edition\Assets\UI\PartialScreens:1005**
`function AddCityStateRowMod( kCityState:table )`

**CQUI_Community-Edition\Assets\UI\PartialScreens:835** (I added some prints it may be near here)
`local kInst :table = AddCityStateRowMod( kCityState );`

Made it work flawlessly:

![image](https://user-images.githubusercontent.com/14226299/46070607-9b32a000-c154-11e8-9b94-191a5b026b6e.png)

**Comment by ihendriks** [comment]
@lautit Thanks a lot for looking into it. I haven't had time to look at it on my PC yet, I'll do so in the weekend.

In the meantime: are you running any official DLC's, and if so, which ones? I've noticed this issue occurs often when you aren't running some of the older official DLC's (Aztec or Australia, for example). The issue sometimes comes from a conflict with another mod, but only when some official DLC is missing.

Maybe it's worth testing your fix with different combinations of official DLC, to see if that resolves the matter for everyone having this issue.

Either way, I'll take a look this weekend as well.

**Comment by lautit** [comment]
@ihendriks good insight. I'm running all(?) the official DLC's, as well as a couple of other mods (Comunity QUI, AI+, JFD's Civs, etc.)

![image](https://user-images.githubusercontent.com/14226299/46137957-13669780-c221-11e8-88b3-aa17c82c4e89.png)

There are a couple of other errors here and there. I think this is more a fix for R&F compatibility rather than all-round compatible.

**Comment by jigbim889** [comment]
When will this be rolled out into the subscription? I am not confident editing LUA.  I have already tried reinstalling the mods.

**Comment by ihendriks** [comment]
@jigbim889 I really don't know. Depends on whether the fix is complete, and when @Azurency gets around to checking it and merging. I'll know more this weekend, probably.

@lautit You have all of those enabled in game as well? Can you also try without any other mods (only official DLC and CQUI).

**Comment by lautit** [comment]
@jigbim889 it shouldn't be a problem editing the LUA, as it is a common text file. Nothing will be broken, do dare! 😁

@ihendriks all enabled in game too, I'll try selectively disabling each one in release order later today or during the weekend and report on the outcome.

**Comment by lautit** [comment]
Well, after some time I managed to finish 2 games using:

- No plugins at all except CQUI
- Only Civilizations DLCs (Aztec, Australian, etc.) but without R&F

Evidence [here](https://imgur.com/a/NPqt4CP). For R&F, it has been tested before and screen is a couple of replies up this one.

What do you say? Are we cool to pull req?

**Comment by ihendriks** [comment]
Yeah you can definitely make a pull request. It'll have to wait until @Azurency is available, because I don't have any actual permission :(

I haven't had any time to look at it myself either, but I've looked at similar issues and it's pretty much the same

**Comment by Masterchief1752** [comment]
Hello. Does this fix still work? I tried swapping all AddCityStateRow to AddCityStateRowMod, but that doesn't seem to work. In fact, the City State menu is just blank. It says I have 1 Enjoy but there's nothing on the menu which leads me to believe I screwed something up. What are the steps for your above response? I'm not a stranger to editing LUA, but I guess I'm a little confused on what needs to be done to refactor outside of finding the above things that I changed. Thank you for any help. 

> I think I got it working. There is a conflict between the function `AddCityStateRow` in CQUI_Community-Edition\Assets\UI\PartialScreens and the one in \DLC\Expansion1\UI\Replacements.
> 
> A simple refactor:
> 
> **CQUI_Community-Edition\Assets\UI\PartialScreens:1005**
> `function AddCityStateRowMod( kCityState:table )`
> 
> **CQUI_Community-Edition\Assets\UI\PartialScreens:835** (I added some prints it may be near here)
> `local kInst :table = AddCityStateRowMod( kCityState );`
> 
> Made it work flawlessly:
> 
> ![image](https://user-images.githubusercontent.com/14226299/46070607-9b32a000-c154-11e8-9b94-191a5b026b6e.png)

**Comment by FlyPhish** [comment]
I'm having the same issue - was running city state wall mod and CQUI - only one city state shows up in the menu and I can't add any envoys

**Comment by GaXyLa** [comment]
@Masterchief1752 @lautit 
Works like a charm.
You have just to edit the function name. So it conflicts not with the DLC version

(Im using only CQUI, no other Mods. Rise & Fall is installed)

@FlyPhish
Look at the post from lautit above

**Comment by IngmarBitter** [comment]
I checked the current (2020.06.27) state of the Assets\UI\PartialScreens\cityStates.lua file. 
It still has 
`function AddCityStateRow( kCityState:table )` which I changed to 
`function AddCityStateRowMod( kCityState:table )`
and 
`local kInst :table = AddCityStateRow( kCityState );` which I changed to 
`local kInst :table = AddCityStateRowMod( kCityState );`

It seems to me the "fix" makes city states appear in the list if it was blank.

In my case the problem is that the city state shows up but clicking on the '>' button does not assign the envoy.
Is there a fix for that?

**Comment by the-m4a** [comment]
@IngmarBitter - are you using other mods?  I haven't been able to reproduce this issue.
Can you post this as an issue in the CivFanatics repo?  https://github.com/civfanatics/CQUI_Community-Edition/issues
And also post your Lua.log and Modding.log files?  (I don't know where they're stored on a Mac, likely near where your save files are, in Logging folder)
